src/pages/layout/Search/Index.tsx

```diff
@@ -201,15 +217,50 @@ export default function HeaderSearch() {
     }
 
     const resultsList = await Promise.all(promises);
-    const wasFound = resultsList.find(
-      (result) =>
-        result?.label?.startsWith("Account") ||
-        result?.label?.startsWith("Object"),
+    const isAccount = resultsList.find((result) =>
+      result?.label?.startsWith("Account"),
+    );
+    const isObject = resultsList.find((result) =>
+      result?.label?.startsWith("Object"),
+    );
+    const isPossiblyDeletedObject = resultsList.find((result) =>
+      result?.label?.startsWith("Deleted Object"),
     );
-    const results = resultsList
-      .filter((result) =>
-        wasFound ? !result?.label?.startsWith("Deleted Object") : true,
-      )
+    const isPossiblyAddress = resultsList.find((result) =>
+      result?.label?.startsWith("Address"),
+    );
+
+    let filteredResults: any[] = [];
+    let result: any;
+    if (isAccount) {
+      const result = resultsList.find((result) =>
+        result?.label?.startsWith("Account"),
+      );
+      filteredResults = [result];
+    } else if (isObject) {
+      const result = resultsList.find((result) =>
+        result?.label?.startsWith("Object"),
+      );
+      filteredResults = [result];
+    } else if (isPossiblyDeletedObject) {
+      const result = resultsList.find((result) =>
+        result?.label?.startsWith("Deleted Object"),
+      );
+      filteredResults = [result];
+    } else if (isPossiblyAddress) {
+      const result = resultsList.find((result) =>
+        result?.label?.startsWith("Address"),
+      );
+      filteredResults = [result];
+    } else {
+      filteredResults = resultsList;
+    }
+    if (filteredResults === undefined || filteredResults === null) {
+      filteredResults = result ? [] : resultsList;
+    }
```

Comment:
`Array.find` returns us the value or `null`, so we are doubling our work here. We can simplify it a bit by renaming the predicates to reflect the item that is potentially be found, then test the existence of those items. You don't need the `switch` style check here, I just find it sometimes cleaner than deep `if else` chains as it implies there will ever only be one match. 

```tsx
  const resultsList = await Promise.all(promises);

  const foundAccount = resultsList.find((r) => r?.label?.startsWith('Account'));
  const foundObject = resultsList.find((r) => r?.label?.startsWith('Object'));
  const foundDeletedObject = resultsList.find((r) => r?.label?.startsWith('Deleted Object'));
  const foundPossibleAddress = resultsList.find((r) => r?.label?.startsWith('Address'));

  // Something besides any
  let filteredResults: any[] = [];
  let result: any;
  
  switch (true) { 
    case Boolean(foundAccount): {
      filteredResults = [foundAccount];
      break;
    }
    case Boolean(foundObject): {
      filteredResults = [foundObject];
      break;
    }
    case Boolean(foundDeletedObject): {
      filteredResults = [foundDeletedObject];
      break;
    }
    case Boolean(foundPossibleAddress): {
      filteredResults = [foundPossibleAddress];
      break;
    }
    default: {
      filteredResults = resultsList;
    }
}
```

Reply:
Okay, this switch is sorcery to me... switch true and then the condition